### PR TITLE
added fastani.net

### PR DESCRIPTION
**Pull request by Blatzar**
based on twist.moe source code

### CODE REVIEW

**Comment by Blatzar** [comment]
Will hopefully fix Codacy Static Code Analysis tomorrow

**Comment by Blatzar** [comment]
image
Example image of fastani status. 

This PR is ready for review.

File: websites/F/fastani.net/presence.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused"
+  });
+
+/**
+ * Get Timestamps
+ * @param {Number} videoTime Current video time seconds
+ * @param {Number} videoDuration Video duration seconds
+ */
```
Comment (ririxidev) [inline websites/F/fastani.net/presence.ts:13]: remove these comments

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/F/fastani.net/presence.ts (hunk 1 above)
Comment (Blatzar) [reply websites/F/fastani.net/presence.ts]: fixed :)

**Review by ririxidev — DISMISSED** [review]

File: websites/F/fastani.net/presence.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    browsingStamp = Math.floor(Date.now() / 1000);
+const urlRegex = /watch\/.*?\/(\d+)\/(\d+)/;
+
+if (lastPlaybackState != playback) {
+  lastPlaybackState = playback;
+  browsingStamp = Math.floor(Date.now() / 1000);
+}
+
+presence.on("UpdateData", async () => {
+  playback = document.querySelector("div.plyr__video-wrapper > video") !== null ? true : false;
```
Comment (Bas950) [inline websites/F/fastani.net/presence.ts:29]: Pretty sure that if statement is not needed.

File: websites/F/fastani.net/presence.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+
+  if (!playback) {
+    presenceData.details = "Browsing...";
+    presenceData.startTimestamp = browsingStamp;
+
+    delete presenceData.state;
+    delete presenceData.smallImageKey;
```
Comment (Bas950) [inline websites/F/fastani.net/presence.ts:41]: Don't delete if it ain't set yet

**Review by Bas950 — CHANGES_REQUESTED** [review]
Support other parts of the website too and not only/watch

**Comment by Blatzar** [comment]
> Support other parts of the website too and not only/watch

I added support for viewing an anime title (looking at episodes and such). I don't really see more exciting presences you can add, there's only home and anime list screens, both with pretty much same content. Example of looking at attack on titan, e.g this screen:
image
image

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Pull request merged**